File: MATLAB code/Run_with_python/convert_pred.py

import sys

from U_Net import convert_pred as convert_pred_py

from keras.models import load_model

if __name__ == "__main__" : 
    model_path = sys.argv[1]
    src_path = sys.argv[2]
    pred_dir = sys.argv[3]
    pattern = sys.argv[4]
    sample_rate = float(sys.argv[5])

    model = load_model(model_path)
    print("\n\nLoaded model from [{}]".format(model_path))

    print("Converting data...", end="")
    convert_pred_py(model, src_path, pred_dir, pattern, sample_rate=sample_rate)
    print("\t--> Done")

File: MATLAB code/Run_with_python/convert_pred.m

% Args
% ------------------------------------------------------------------------------------- %
% 1. model_path     :   path where model was saved.                                     %
% 2. src_path       :   path where data exists.                                         %
% 3. pred_dir       :   path where converted data will be stored.                       %
%                                                                                       %
% 4. pattern        :   file pattern to use data.                                       %
%                       if pattern="*.wav", it will use every .wav files as data        %
% 5. sample_rate    :   sample rate to resolve data.                                    %
% ------------------------------------------------------------------------------------- %

function convert_pred(model_path, src_path, pred_dir, pattern, sample_rate)
    
    args = model_path + " " + src_path + " " + pred_dir + " " + pattern + " " + sample_rate;
    args = "convert_pred.py " + args;
    pyrunfile(args)
end